eclipse-che/che-operator
# feat: define routing paths for che-server to improve return http codes

…xisting paths

Signed-off-by: Michal Vala <mvala@redhat.com>



### What does this PR do?
Define routing paths that should be routed into che-server. This fixes the issue when any non-existing path is redirected into che-server, instead of returning 404.

TODO: 
 - what happens if open url of stopped devworkspace?
   - with current implementation => "504 Gateway Time-out" because route configmaps are still there.
 - can we route `/` directly to dashboard ?

### Screenshot/screencast of this PR

![20210916_191655_808x133_scrot](https://user-images.githubusercontent.com/1160903/133656411-63136a22-bede-4ab5-a612-31f6de29cbd8.png)


### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20148

### How to test this PR?


image `quay.io/mvala/che-operator:gh20148-404`

1. deploy with single-host strategy with che-operator image ^
2. login into che and try to open any non-existing path, like `<che-url>/blabol`, it should just return 404. Or open url of stopped workspace.


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
